### DISCUSSION TITLE
Conversion of temp/weights

### DISCUSSION THREAD

**Discussion by minkshaman** [mealie-recipes/mealie#2766] [Feature Request]
### Discussed in https://github.com/mealie-recipes/mealie/discussions/2730

<div type='discussions-op-text'>

<sup>Originally posted by **minkshaman** November 18, 2023</sup>
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

I've had many recipes successfully imported from URL, but have found they are in imperial weights and fahrenheit.

I have to manually adjust these when entering the recipe.

### Please provide a concise description of the feature that would resolve your issue.

It would be very useful to have a function that can scan through all text (ingredients and method) and alert if it finds something that appears to be either a temperature or a measurement, and then prompt for if it should be metric/Celsius or imperial/fahrenheit.

**Comment by boc-the-git** [comment]
Can you provide some sample URLs for this @minkshaman? I'm keen to have some reference recipes.

**Comment by boc-the-git** [comment]
(to be clear, I'm not starting development work on this, just wanting to understand the scenario)

**Comment by minkshaman** [comment]
All good.

https://www.allrecipes.com/recipe/42176/all-american-roast-beef/

This is a pretty good example.
So in this, I’d like it to prompt and suggest 450g instead of 1lb (or the exact to 2 places? Whatever’s easiest.)

For temps, it should look at the temperature and prompt me if it needs to be C or F and change the temperature accordingly.

if you wanted to be fancy, you could probably put something that looks for Imperial weights and use that to prompt? I don’t know if it’s consistent enough though.

**Comment by LayerCakeMakes** [comment]
I'd also appreciate a conversion of units, possibly with a user setting for preffered units kg vs lb oz etc.
But this probably would also require automatic recoqnition of units in the first place.
Additionally some way to manage the units that are currently known would be great too.

**Comment by pekvasnovsky** [comment]
I also think this feature would be nice
but I would also want to add that it would be neat to have the conversion more powerful e.g.:

300ml oil -> 276 grams

and similar stuff

I feel like this would require to have a table of volumetric mass densities for everything

this is kinda possible and it would also have to allow the user to edit this table because maybe some ingredients have different density in the region the user is in

it should also work in a way like you want to have "american cups" from grams of particular ingredient

so the default values in the recipe should be the common human friendly values but then one should be easily able to convert to desired units so the preparation requires less measuring devices so it is faster, less stuff gets dirty
